### ISSUE TITLE
Support dbFetch(n = 50000) instead of multiple queries

### ISSUE THREAD

**Issue by krlmlr** [cboettig/arkdb#7]
as an option (or perhaps the default behavior)?

```r
res <- dbSendQuery(con, "SELECT * FROM ...")
while (TRUE) {
  data <- dbFetch(res, n = 50000)
  if (nrow(data) == 0) break
  readr::write_tsv(data, ...)
}
dbClearResult(res)
```

Many DBI backends will not reserve client-side memory for the full result set after `dbSendQuery()`. Also, the result won't be mixed up when other clients alter the table during export.

**Comment by krlmlr** [comment]
If a `dbSendQuery()` immediately transfers every thing to the client, the current solution works better. But I'm not aware of DBI backends that would do that.

I'm preparing a DBI presentation, I might be able to take a look. But there seems to be potential for conflicts with #5.

**Comment by cboettig** [comment]
@krlmlr Re conflicts with #5, don't let that hold you back if you want to PR this; I can deal with that later.  (Or when I have a chance I'll try and just implement what you have above as the new default behavior).  

(in #5, looks like @richfitz is advising me to implement something along those lines, so to speak, rather than intending I merge that directly. I do think #5 looks like a solid building block so I'm tempted to merge and deal with the follow-up later...)

**Comment by cboettig** [comment]
implemented in #8
